src/game/gameflow.c

```diff
@@ -1273,6 +1293,11 @@ GameFlow_InterpretSequence(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
             return GF_EXIT_TO_TITLE;
 
         case GFS_EXIT_TO_LEVEL:
+            if (((int32_t)seq->data & ((1 << 6) - 1))
+                    >= g_GameFlow.first_bonus_num
+                && !g_GameInfo.bonus_level_unlock) {
+                return GF_EXIT_TO_TITLE;
+            }
             return GF_START_GAME
```

Comment:
Maybe we can do `int32_t next_level = (int32_t)(intptr_t)seq->data & ((1 << 6) - 1);` just once and use it for the condition and the later return statement.

Reply:
Ya good point.

Also I'm sure there's some reason it's done this way, but what's the difference between `seq->data & ((1 << 6) - 1)` and `level_num` that gets passed into `GameFlow_InterpretSequence`?

Reply:
Found a bug while testing this. For gameflows without a bonus level, `g_GameFlow.first_bonus_num` is `-1`. So the game always returns to title after the first level. LMK if you think the solution I added works.